docs/proposals/control-data-plane-split/README.md

```diff
@@ -52,10 +52,8 @@ Whenever a user creates a Gateway resource, the control plane will provision an
 - Both deployments should have read only filesystems.
 - Both deployments should have the minimal permissions required to perform their functions.
 - The nginx deployment should be configurable via the helm chart.
-  - Downside of this is that these options will apply to all nginx instances.
-  - We could introduce a CRD, but where would it attach? We already have NginxProxy which controls dynamic data plane configuration, and this may eventually attach to the Gateway instead of just the GatewayClass. Would a Deployment configuration fit in there, and would it be dynamic? That would require us to completely redeploy nginx if a user changes those settings.
-  - We could start with the helm chart option, and rely on user feedback to see if we need to get more granular.
-  - This could also involve creating a ConfigMap that the control plane consumes on startup and contains all nginx Deployment/Daemonset configuration, including NGINX Plus usage configuration.
+  - In the future, we could support adding all nginx Deployment/Service configuration to the NginxProxy CRD, and support that CRD at the Gateway level.
+  - For now, we can allow a user to set all of this in the helm chart on installation. The helm chart could have a list of `gateways` where they provide the name of a Gateway and associated nginx Deployment/Service configuration options. This would then be populated in a ConfigMap that the control plane consumes on startup and uses when deploying nginx instances. A user could update the ConfigMap at runtime to change or add any configuration for existing or future nginx Deployments/Services.
```

Comment:
I think there are a few problems with this approach:
- It assumes that one person will create and know about all Gateways
- ConfigMaps can't be validated by schemas
- It doesn't leverage a Gateway API extension point and is difficult to discover.

Reply:
I agree, and this was mainly an idea to just get something in place, because I suspect that adding this to NginxProxy AND supporting that CRD at the Gateway level is going to take some time.

Though if we want to avoid breaking changes in the future by removing the ConfigMap in favor of NginxProxy, we could just disallow users to update the ConfigMap after the fact, and make it only configurable at install time.

Although for manifest users, they would have to switch from using the ConfigMap to NginxProxy. Helm users the change could be transparent behind the scenes.

Reply:
If we do use the CRD approach though, I think it has to live in NginxProxy since we can only support one CRD, and I feel that NginxProxy *should* probably be supported at the Gateway level.

Reply:
I realized that we can't have a global nginx Deployment/Service config because each Service is going to have its own `loadBalancerIP` field and things like that.

Unless we find a way to allow some things to be defined globally and some things to be specific to each deployment/service.

Reply:
One downside to the new approach is that adding a new field to the deployment isn't as easy as just adding it to the helm chart like today. It also needs to be added to the CRD and Go code. This is probably unavoidable though.

The more I think about this, the more I think we just need to support NginxProxy at the Gateway level so we avoid future breaking changes when we change the method from ConfigMap to CRD.